benchmark/torch/a2c/model.py

```diff
@@ -0,0 +1,91 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+import parl
+
+
+class ActorCritic(parl.Model):
+    def __init__(self, act_dim):
+        super(ActorCritic, self).__init__()
+        self.conv1 = nn.Conv2d(
+            in_channels=4, out_channels=32, kernel_size=8, stride=4, padding=2)
+        self.conv2 = nn.Conv2d(
+            in_channels=32,
+            out_channels=64,
+            kernel_size=4,
+            stride=2,
+            padding=2)
+        self.conv3 = nn.Conv2d(
+            in_channels=64,
+            out_channels=64,
+            kernel_size=3,
+            stride=1,
+            padding=1)
+        self.fc = nn.Linear(7744, 512)
+
+        self.fc_pi = nn.Linear(512, act_dim)
+        self.fc_v = nn.Linear(512, 1)
+
+    def policy(self, x, softmax_dim=1):
+        x = x / 255.0
+        x = F.relu(self.conv1(x))
+        x = F.relu(self.conv2(x))
+        x = F.relu(self.conv3(x))
+
+        x = x.view(-1, self.num_flat_features(x))
+        x = F.relu(self.fc(x))
+
+        logits = self.fc_pi(x)
+        prob = F.softmax(logits, dim=softmax_dim)
+
+        return prob
+
+    def value(self, x):
+        x = x / 255.0
+        x = F.relu(self.conv1(x))
+        x = F.relu(self.conv2(x))
+        x = F.relu(self.conv3(x))
+
+        x = x.view(-1, self.num_flat_features(x))
+        x = F.relu(self.fc(x))
+        values = self.fc_v(x)
+
+        return values
+
+    def policy_and_value(self, x, softmax_dim=1):
+        x = x / 255.0
+        x = F.relu(self.conv1(x))
+        x = F.relu(self.conv2(x))
+        x = F.relu(self.conv3(x))
+
+        x = x.view(-1, self.num_flat_features(x))
+        x = F.relu(self.fc(x))
+
+        values = self.fc_v(x)
+        logits = self.fc_pi(x)
+        prob = F.softmax(logits, dim=softmax_dim)
+
+        return prob, values
+
+    def num_flat_features(self, x):
```

Comment:
It is possible to replace this function with the API here: https://pytorch.org/docs/stable/_modules/torch/nn/modules/flatten.html ?